components/Token/Media.tsx

```diff
@@ -55,25 +33,131 @@ const TokenMedia: FC<{
+        // we know the mimetype
+        contentsToDisplay.push({
+          url: unlockedContent.url,
+          mimetype: unlockedContent.mimetype,
+        })
+      } else {
+        // we don't know the mimetype, start with image and then video
+        contentsToDisplay.push({
+          url: unlockedContent.url,
+          fallbackMimetype: 'image/*',
+        })
+        contentsToDisplay.push({
+          url: unlockedContent.url,
+          fallbackMimetype: 'video/*',
+        })
+      }
+    }
+    if (animationUrl) {
+      contentsToDisplay.push({
+        url: animationUrl,
+        fallbackMimetype: 'video/*',
+      })
+      contentsToDisplay.push({
+        url: animationUrl,
+        fallbackMimetype: 'image/*',
+      })
+    }
+    contentsToDisplay.push({
+      url: imageUrl,
+      fallbackMimetype: 'image/*',
+    })
+    contentsToDisplay.push({
+      url: imageUrl,
+      fallbackMimetype: 'video/*',
+    })
+    return contentsToDisplay
+  }, [animationUrl, imageUrl, unlockedContent])
+  console.log('contentsToDisplay', contentsToDisplay)
+
+  // try to display the first element of contentsToDisplay
+  const [currentContentToDisplay, setCurrentContentToDisplay] =
+    useState<number>(0)
+
+  // reset current content to display when imageUrl, animationUrl, or unlockedContent change
+  useEffect(() => setCurrentContentToDisplay(0), [contentsToDisplay])
+
+  // on error, switch to the next content to display
+  const onError = useCallback(() => {
+    console.warn('error onError')
+    setCurrentContentToDisplay((x) => {
+      console.warn('error displaying content', x)
+      return x + 1 // increase current content to display by one
+    })
+  }, [])
+
+  // element to display
+  const toDisplay = useMemo(
+    () => contentsToDisplay.at(currentContentToDisplay),
+    [contentsToDisplay, currentContentToDisplay],
+  )
+
+  // mimetype of the element to display
+  const [mimetype, setMimetype] = useState<string | undefined>(
+    toDisplay?.mimetype,
   )
 
-  const [imageError, setImageError] = useState(false)
-  const [videoError, setVideoError] = useState(false)
+  // fetch mimetype if not set
+  useEffect(() => {
+    // set mimetype if defined in toDisplay
+    if (toDisplay?.mimetype) return setMimetype(toDisplay.mimetype)
+
+    // reset mime type
+    setMimetype(undefined)
+
+    // check no execution cases
+    if (!toDisplay) return
+    if (
+      !toDisplay.url.startsWith('http') ||
+      !toDisplay.url.startsWith('https')
+    ) {
+      // do not try to fetch mime type if url is not http or https
+      return
+    }
 
-  const onImageError = useCallback(() => setImageError(true), [])
-  const onVideoError = useCallback(() => setVideoError(true), [])
+    // fetch mime type
+    fetch(toDisplay.url, {
+      method: 'HEAD',
+      // TODO: add request cancelation
+    })
+      .then((response) => {
+        const mimetype = response.headers.get('Content-Type')
```

Comment:
Should also check the lowercased version of the header:
```suggestion
        const mimetype = response.headers.get('Content-Type') || response.headers.get('content-type')
```